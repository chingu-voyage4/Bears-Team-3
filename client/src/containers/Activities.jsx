import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import ExpansionPanel, {
  ExpansionPanelSummary,
  ExpansionPanelDetails,
} from 'material-ui/ExpansionPanel';
import Typography from 'material-ui/Typography';
import ExpandMoreIcon from 'material-ui-icons/ExpandMore';
import { Link } from 'react-router-dom';

import * as actions from '../actions';

const getRandomNum = max => Math.floor(Math.random() * Math.floor(max));
const goals = [
  'To be the very best',
  'To become the hokage',
  'Chillax with my bros',
];
const studyPlan = ['Do this first', 'Do that first', 'Do nothing'];
const currentCourse = [
  'Being a Boss 101',
  'Advanced Jutsu',
  "Professor Oak's Tutorial",
];

class Activities extends Component {
  state = { isAuthenticated: false };

  handleClick = () => {
    this.props.addActivity({
      title: 'My Project',
      url: 'www.notreal.com',
      activity: 'Basic Project',
    });
    this.props.fetchActivities(this.props.userPage._id);
  };

  handleDelete = id => {
    this.props.deleteActivity(id);
    this.props.fetchActivities(this.props.userPage._id);
  };

  handleUpdate = id => {
    const urls = [
      'www.yamama.com',
      'https://whosthere.com',
      'http://speedstudy.herokuapp.com',
    ];
    const titles = ['The Greatest Project', 'Meh', 'I Did This'];
    const projectType = [
      'Basic Project',
      'Physical Activity',
      'Tutorial Course',
    ];

    const getRandomNum = max => Math.floor(Math.random() * Math.floor(max));

    this.props.modifyActivity(id, {
      url: urls[getRandomNum(2)],
      title: titles[getRandomNum(2)],
      activity: projectType[getRandomNum(2)],
    });
    this.props.fetchActivities(this.props.userPage._id);
  };

  addProgress = () => {
    this.props.addProgressData({
      goal: goals[getRandomNum(2)],
      studyPlan: studyPlan[getRandomNum(2)],
      currentCourse: currentCourse[getRandomNum(2)],
    });
  };

  modifyProgress = () => {
    this.props.modifyProgressData({
      goal: goals[getRandomNum(2)],
      studyPlan: studyPlan[getRandomNum(2)],
      currentCourse: currentCourse[getRandomNum(2)],
    });
  };

  deleteProgress = () => {
    this.props.deleteProgressData();
  };

  deleteUser = () => {
    this.props.deleteUser();
  };

  checkAuth = () => {
    const { match: { params }, userAuthenticated: { userName } } = this.props;

    if (params.username === userName) {
      this.setState(prevState => {
        if (!prevState.isAuthenticated) return { isAuthenticated: true };
        return;
      });
      return true;
    }

    this.setState(prevState => {
      if (prevState.isAuthenticated) return { isAuthenticated: false };
      return;
    });
    return false;
  };

  async componentDidMount() {
    try {
      await this.props.fetchUserInfo(this.props.match.params.username);
      this.props.fetchActivities(this.props.userPage._id);
      this.props.fetchProgressData(this.props.userPage._id);
    } catch (err) {
      this.props.history.push(`/404/${this.props.match.params.username}`);
    }
  }

  componentDidUpdate() {
    this.checkAuth();
  }

  renderActivities = () => {
    return this.props.activities.map((action, i) => {
      const { title, points, dateCompleted, activity, url, _id } = action;
      return (
        <ExpansionPanel key={i}>
          <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
            <Typography style={{ width: '100%' }}>
              <b style={{ float: 'left' }}>{title}</b>
              <b style={{ float: 'right' }}>{points} points</b>
            </Typography>
          </ExpansionPanelSummary>
          <ExpansionPanelDetails style={{ display: 'block' }}>
            <p>
              Completed {activity} on{' '}
              {new Date(dateCompleted).toLocaleDateString()}
            </p>
            <p>{url}</p>
            {this.state.isAuthenticated && (
              <React.Fragment>
                <button onClick={() => this.handleDelete(_id)}>
                  Delete Activity
                </button>
                <button onClick={() => this.handleUpdate(_id)}>
                  Update Activity
                </button>
              </React.Fragment>
            )}
          </ExpansionPanelDetails>
        </ExpansionPanel>
      );
    });
  };

  render() {
    const { activities, userPage } = this.props;

    if (userPage) {
      const {
        currentCourse,
        goal,
        studyPlan,
        totalPoints,
        userName,
      } = userPage;
      return (
        <React.Fragment>
          <h2>{userName}</h2>
          <p>Total points: {totalPoints}</p>
          <div>
            <div>Current Course: {currentCourse}</div>
            <div>Goal: {goal}</div>
            <div>Study Plan: {studyPlan}</div>
          </div>
          <hr />
          {this.state.isAuthenticated && (
            <React.Fragment>
              <Link to="/activity/new">Add Activity</Link>
              <Link to="/progress/new">Add Progress</Link>
              <button onClick={this.modifyProgress}>Edit Progress</button>
              <button onClick={this.deleteProgress}>Delete Progress</button>
              <button onClick={this.deleteUser}>Delete Account</button>
            </React.Fragment>
          )}
          <hr />
          {activities && (
            <React.Fragment>
              <h2>Activities</h2>
              {this.renderActivities()}
            </React.Fragment>
          )}
        </React.Fragment>
      );
    } else {
      return <p>Loading...</p>;
    }
  }
}

const mapStateToProps = state => ({
  userAuthenticated: state.authReducer,
  activities: state.activities,
  userPage: state.userPage,
});

const mapDispatchToProps = dispatch => {
  return bindActionCreators({ ...actions }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(Activities);
