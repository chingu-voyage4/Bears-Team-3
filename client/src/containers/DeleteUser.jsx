import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Button from 'material-ui/Button';

import { deleteUser } from '../actions';

class DeleteUser extends Component {
  render() {
    return (
      <div>
        <h2>Delete Your Account</h2>
        <p>This will permanently remove all of your data</p>
        <p>Are you sure?</p>
        {this.renderRedirect()}
        <Button
          variant="raised"
          color="primary"
          onClick={() => {
            this.props.deleteUser();
          }}
        >
          Yes, Delete My Account
        </Button>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators({ deleteUser }, dispatch);
};

export default connect(null, mapDispatchToProps)(DeleteUser);
