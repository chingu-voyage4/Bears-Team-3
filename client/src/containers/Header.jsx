import React, { Component } from 'react';
import { connect } from 'react-redux';

import * as actions from '../actions';

class Header extends Component {
  constructor(props) {
    super(props);

    this.handleLogin = this.handleLogin.bind(this);
  }

  handleLogin() {
    this.props.login();
  }

  renderContent() {
    switch (this.props.auth) {
      case null:
        return;
      case false:
        return (
          <button
            type="button"
            className="nav__login"
            onClick={this.handleLogin}
          >
            Log In With Github
          </button>
        );
      default:
        return (
          <div className="nav__logout">
            <a href="/api/logout">Logout</a>
          </div>
        );
    }
  }

  render() {
    return <nav className="nav">{this.renderContent()}</nav>;
  }
}

const mapStateToProps = state => ({
  auth: state.authReducer,
});

export default connect(mapStateToProps, actions)(Header);
