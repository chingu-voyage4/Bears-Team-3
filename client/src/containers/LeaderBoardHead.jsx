import React from 'react';
import PropTypes from 'prop-types';
import {
  TableCell,
  TableHead,
  TableRow,
  TableSortLabel,
} from 'material-ui/Table';

import Tooltip from 'material-ui/Tooltip';

const columnData = [
  {
    id: 'name',
    numeric: false,
    disablePadding: true,
    label: 'User Name',
  },
  {
    id: 'totalPoints',
    numeric: true,
    disablePadding: false,
    label: 'Total Points',
  },
];

export class LeaderBoardHead extends React.Component {
  createSortHandler = property => event => {
    this.props.onRequestSort(event, property);
  };

  render() {
    const { order, orderBy } = this.props;

    return (
      <TableHead>
        <TableRow>
          {columnData.map(column => {
            return (
              <TableCell
                key={column.id}
                numeric={column.numeric}
                padding={'default'}
                sortDirection={orderBy === column.id ? order : false}
              >
                <Tooltip
                  title="Sort"
                  placement={column.numeric ? 'bottom-end' : 'bottom-start'}
                  enterDelay={300}
                >
                  <TableSortLabel
                    active={orderBy === column.id}
                    direction={order}
                    onClick={this.createSortHandler(column.id)}
                  >
                    {column.label}
                  </TableSortLabel>
                </Tooltip>
              </TableCell>
            );
          }, this)}
        </TableRow>
      </TableHead>
    );
  }
}

LeaderBoardHead.propTypes = {
  onRequestSort: PropTypes.func.isRequired,
  order: PropTypes.string.isRequired,
  orderBy: PropTypes.string.isRequired,
  rowCount: PropTypes.number.isRequired,
};

export default LeaderBoardHead;
