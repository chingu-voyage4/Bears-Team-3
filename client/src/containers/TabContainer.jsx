import React from 'react';
import PropTypes from 'prop-types';
import Typography from 'material-ui/Typography';

function TabContainer(props) {
  return (
    <Typography component="div" style={{ marginBottom: '0px' }}>
      {props.children}
    </Typography>
  );
}

TabContainer.propTypes = {
  children: PropTypes.node,
};

export default TabContainer;
