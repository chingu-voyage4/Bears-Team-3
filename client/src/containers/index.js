export { default as Home } from './Home';
export { default as Header } from './Header';
export { default as User } from './User';
export { default as UserActivities } from './UserActivities';
export { default as UserGoals } from './UserGoals';
export { default as UserStudyPlan } from './UserStudyPlan';
export { default as Activities } from './Activities';
export { default as LeaderBoard } from './LeaderBoard';
export { default as LeaderBoardHead } from './LeaderBoardHead';
export { default as DeleteUser } from './DeleteUser';
