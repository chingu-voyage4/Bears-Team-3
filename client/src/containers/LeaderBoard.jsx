import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import Table, { TableBody, TableCell, TableRow } from 'material-ui/Table';
import Paper from 'material-ui/Paper';
import axios from 'axios';

import LeaderBoardHead from './LeaderBoardHead';

const styles = theme => ({
  root: {
    width: '80%',
    [theme.breakpoints.up('sm')]: {
      width: '50%',
    },
    marginTop: theme.spacing.unit * 3,
    padding: `${theme.spacing.unit / 2}px ${theme.spacing.unit * 3}px ${theme
      .spacing.unit / 2}px ${theme.spacing.unit * 3}px`,
    '&:last-child': {
      paddingRight: theme.spacing.unit * 3,
    },
  },
  table: {
    minWidth: 50,
  },
  tableWrapper: {
    overflowX: 'auto',
  },
});

class LeaderBoard extends Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      order: 'asc',
      orderBy: 'totalPoints',
      selected: [],
      data: [].sort((a, b) => (a.totalPoints < b.totalPoints ? -1 : 1)),
    };
  }

  componentDidMount = () => {
    axios.get('/api/leaderboard').then(
      function(res) {
        this.setState({ data: res.data });
      }.bind(this)
    );
  };

  handleRequestSort = (event, property) => {
    const orderBy = property;
    let order = 'desc';

    if (this.state.orderBy === property && this.state.order === 'desc') {
      order = 'asc';
    }

    let data =
      order === 'desc'
        ? this.state.data.sort((a, b) => (b[orderBy] < a[orderBy] ? -1 : 1))
        : this.state.data.sort((a, b) => (a[orderBy] < b[orderBy] ? -1 : 1));

    this.setState({ data, order, orderBy });
  };

  render() {
    const { classes } = this.props;
    const { data, order, orderBy } = this.state;

    return (
      <div className="table">
        <Paper className={classes.root}>
          <div className={classes.tableWrapper}>
            <Table className={classes.table}>
              <LeaderBoardHead
                order={order}
                orderBy={orderBy}
                onRequestSort={this.handleRequestSort}
                rowCount={data.length}
              />
              <TableBody>
                {data.map(n => {
                  return (
                    <TableRow hover tabIndex={-1} key={n._id}>
                      <TableCell className={classes.root}>
                        <Link to={{ pathname: `/users/${n.userName}` }}>
                          {n.userName}
                        </Link>
                      </TableCell>
                      <TableCell className={classes.root} numeric>
                        {n.totalPoints}
                      </TableCell>
                    </TableRow>
                  );
                })}
              </TableBody>
            </Table>
          </div>
        </Paper>
      </div>
    );
  }
}

LeaderBoard.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(LeaderBoard);
