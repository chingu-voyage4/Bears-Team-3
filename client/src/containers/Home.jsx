import React from 'react';
import LeaderBoard from './LeaderBoard';

const Home = ({ totalPoints }) => (
  <div>
    <h2>Home Page</h2>
    <LeaderBoard />
  </div>
);

export default Home;
