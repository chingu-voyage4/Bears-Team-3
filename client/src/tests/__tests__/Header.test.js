import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './../../containers/Header';

it('renders without crashing', () => {
  const classes = {
    test: test,
  };

  const snap = shallow(<Header classes={classes} />);
  expect(snap).toMatchSnapshot();
});
