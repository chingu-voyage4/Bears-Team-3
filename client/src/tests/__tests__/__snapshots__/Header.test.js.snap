// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header handleMenu should set anchorEl to event.currentTarget 1`] = `
<div
  className={[Function]}
>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)>
      <IconButton
        aria-label="Book"
        className={[Function]}
        color="inherit"
        component={[Function]}
        to="/"
      >
        <pure(ImportContacts) />
      </IconButton>
      <Typography
        className={[Function]}
        color="inherit"
        variant="title"
      >
        Speedstudy
      </Typography>
      <div>
        <IconButton
          aria-haspopup="true"
          aria-owns="menu-appbar"
          color="inherit"
          onClick={[Function]}
        >
          <WithStyles(Avatar)
            alt="User Avatar"
          />
        </IconButton>
        <WithStyles(Menu)
          anchorEl={
            Object {
              "value": "<button tabindex=\\"0\\" class=\\"MuiButtonBase-root-47 MuiIconButton-root-41 MuiIconButton-colorInherit-42\\" type=\\"button\\" id=\\"avatar\\" aria-haspopup=\\"true\\" aria-owns=\\"menu-appbar\\"><span class=\\"MuiIconButton-label-46\\"><div class=\\"MuiAvatar-root-87\\"><img alt=\\"User Avatar\\" src=\\"https://avatars2.githubusercontent.com/u/14152877?v=4\\" class=\\"MuiAvatar-img-89\\"></div></span><span class=\\"MuiTouchRipple-root-56\\"></span></button>",
            }
          }
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          id="menu-appbar"
          onClose={[Function]}
          open={true}
          transformOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
        >
          <MenuItem>
            Logout
          </MenuItem>
          <MenuItem
            component={[Function]}
            to={
              Object {
                "pathname": "/undefined",
              }
            }
          >
            My Page
          </MenuItem>
        </WithStyles(Menu)>
      </div>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;

exports[`Header renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    classes={
      Object {
        "flex": [Function],
        "menuButton": [Function],
        "root": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(AppBar)
        position="static"
      >
        <WithStyles(Toolbar)>
          <IconButton
            aria-label="Book"
            className={[Function]}
            color="inherit"
            component={[Function]}
            to="/"
          >
            <pure(ImportContacts) />
          </IconButton>
          <Typography
            className={[Function]}
            color="inherit"
            variant="title"
          >
            Speedstudy
          </Typography>
          <div>
            <WithStyles(Button)
              color="inherit"
              onClick={[Function]}
            >
              Login With Github
            </WithStyles(Button)>
          </div>
        </WithStyles(Toolbar)>
      </WithStyles(AppBar)>,
      "className": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <IconButton
            aria-label="Book"
            className={[Function]}
            color="inherit"
            component={[Function]}
            to="/"
          >
            <pure(ImportContacts) />
          </IconButton>
          <Typography
            className={[Function]}
            color="inherit"
            variant="title"
          >
            Speedstudy
          </Typography>
          <div>
            <WithStyles(Button)
              color="inherit"
              onClick={[Function]}
            >
              Login With Github
            </WithStyles(Button)>
          </div>
        </WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <IconButton
              aria-label="Book"
              className={[Function]}
              color="inherit"
              component={[Function]}
              to="/"
            >
              <pure(ImportContacts) />
            </IconButton>,
            <Typography
              className={[Function]}
              color="inherit"
              variant="title"
            >
              Speedstudy
            </Typography>,
            <div>
              <WithStyles(Button)
                color="inherit"
                onClick={[Function]}
              >
                Login With Github
              </WithStyles(Button)>
            </div>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Book",
              "children": <pure(ImportContacts) />,
              "className": [Function],
              "color": "inherit",
              "component": [Function],
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Speedstudy",
              "className": [Function],
              "color": "inherit",
              "variant": "title",
            },
            "ref": null,
            "rendered": "Speedstudy",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <WithStyles(Button)
                color="inherit"
                onClick={[Function]}
              >
                Login With Github
              </WithStyles(Button)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login With Github",
                "color": "inherit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Login With Github",
              "type": [Function],
            },
            "type": "div",
          },
          undefined,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(AppBar)
          position="static"
        >
          <WithStyles(Toolbar)>
            <IconButton
              aria-label="Book"
              className={[Function]}
              color="inherit"
              component={[Function]}
              to="/"
            >
              <pure(ImportContacts) />
            </IconButton>
            <Typography
              className={[Function]}
              color="inherit"
              variant="title"
            >
              Speedstudy
            </Typography>
            <div>
              <WithStyles(Button)
                color="inherit"
                onClick={[Function]}
              >
                Login With Github
              </WithStyles(Button)>
            </div>
          </WithStyles(Toolbar)>
        </WithStyles(AppBar)>,
        "className": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <IconButton
              aria-label="Book"
              className={[Function]}
              color="inherit"
              component={[Function]}
              to="/"
            >
              <pure(ImportContacts) />
            </IconButton>
            <Typography
              className={[Function]}
              color="inherit"
              variant="title"
            >
              Speedstudy
            </Typography>
            <div>
              <WithStyles(Button)
                color="inherit"
                onClick={[Function]}
              >
                Login With Github
              </WithStyles(Button)>
            </div>
          </WithStyles(Toolbar)>,
          "position": "static",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <IconButton
                aria-label="Book"
                className={[Function]}
                color="inherit"
                component={[Function]}
                to="/"
              >
                <pure(ImportContacts) />
              </IconButton>,
              <Typography
                className={[Function]}
                color="inherit"
                variant="title"
              >
                Speedstudy
              </Typography>,
              <div>
                <WithStyles(Button)
                  color="inherit"
                  onClick={[Function]}
                >
                  Login With Github
                </WithStyles(Button)>
              </div>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Book",
                "children": <pure(ImportContacts) />,
                "className": [Function],
                "color": "inherit",
                "component": [Function],
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Speedstudy",
                "className": [Function],
                "color": "inherit",
                "variant": "title",
              },
              "ref": null,
              "rendered": "Speedstudy",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <WithStyles(Button)
                  color="inherit"
                  onClick={[Function]}
                >
                  Login With Github
                </WithStyles(Button)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login With Github",
                  "color": "inherit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Login With Github",
                "type": [Function],
              },
              "type": "div",
            },
            undefined,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
